controllers/openstackdataplanerole_controller.go

```diff
@@ -407,3 +411,13 @@ func resolveAnsibleVars(node *dataplanev1beta1.NodeSection, host *ansible.Host,
 
 	return nil
 }
+
+// checkDeployProtectionAnnotation checks to see if the node has the deploy protection annotation set. If this is set,
+// we will block the execution of Ansible tasks until it is cleared.
+func (r *OpenStackDataPlaneRoleReconciler) checkDeployProtectionAnnotation(instance *dataplanev1beta1.OpenStackDataPlaneRole, annotation map[string]string) bool {
+	for k := range instance.ObjectMeta.GetAnnotations() {
+		_, keyExists := annotation[k]
+		return keyExists
```

Comment:
ditto (makes sense to move it to some shared place?)

Reply:
It will always need to accept a different type for the instance, ie `*dataplanev1beta1.OpenStackDataPlaneRole` `*dataplanev1beta1.OpenStackDataPlaneNode`, `*dataplanev1beta1.OpenStackDataPlane`. Not sure I can generic out those Types to put a single function in a common place.

Reply:
These have now been moved @bogdando into common.go. I was trying to use Go generics, but @gibizer pointed out that I can just use `client.Object` since all of the OSDP types will satisfy that interface. So that allowed for easier common functions:
https://github.com/openstack-k8s-operators/dataplane-operator/pull/217/files#diff-482605cf1d0af2d53f36083c5b246016c84e00e3c34dedd05ebad6c8cec51dc8R10-R36